test/asymmetric_test.dart

```diff
@@ -0,0 +1,117 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'package:test/expect.dart' hide throws;
+
+import '../lib/realm.dart';
+import 'test.dart';
+
+part 'asymmetric_test.g.dart';
+
+@RealmModel(ObjectType.asymmetricObject)
+class _Asymmetric {
+  @PrimaryKey()
+  @MapTo('_id')
+  late ObjectId id;
+
+  late List<_Embedded> embeddedObjects;
+}
+
+@RealmModel(ObjectType.embeddedObject)
+class _Embedded {
+  late int value;
+  late RealmValue any;
+  _Symmetric? symmetric;
+}
+
+@RealmModel()
+class _Symmetric {
+  @PrimaryKey()
+  @MapTo('_id')
+  late ObjectId id;
+}
+
+Future<void> main([List<String>? args]) async {
+  await setupTests(args);
+
+  Future<Realm> getSyncRealm(AppConfiguration config) async {
+    return await (AppConfiguration config, List<SchemaObject> schemas) async {
+      final app = App(config);
+      final user = await getAnonymousUser(app);
+      final realmConfig = Configuration.flexibleSync(user, schemas);
+      return getRealm(realmConfig);
+    }(config, [Asymmetric.schema, Embedded.schema, Symmetric.schema]);
+  }
+
+  baasTest('Asymmetric objects die even before upload', (config) async {
+    final realm = await getSyncRealm(config);
+    realm.syncSession.pause();
+
+    final oid = ObjectId();
+    realm.write(() {
+      realm.ingest(Asymmetric(oid, embeddedObjects: [1, 2, 3].map(Embedded.new)));
+    });
+
+    // Find & query on an Asymmetric object is compile time error, but you can cheat with dynamic
+    expect(realm.dynamic.find('Asymmetric', oid), null); // should this throw instead?
```

Comment:
On the comment I think this should behave as the case for not found `RealmObject`.

Reply:
I'll remove question